service-handler/service-handler.go

```diff
@@ -0,0 +1,80 @@
+package servicehandler
+
+import (
+	json "encoding/json"
+	xml "encoding/xml"
+	"log"
+	"net/http"
+
+	getapi "thien.com/get-api"
+	responsebuilder "thien.com/response_builder"
+)
+
+const (
+	contentJson = "application/json"
+	contentXml  = "application/xml"
+)
+
+type ResultWriter struct {
+	result []byte
+}
+
+func (resWriter *ResultWriter) Write(p []byte) (n int, err error) {
+	resWriter.result = p
+	return len(p), nil
+}
+
+type serviceHandlerImpl struct {
+	builder     *responsebuilder.ResponseBuilderImpl
+	resWriter   *ResultWriter
+	contentType string
+}
+
+func Get(isJson bool) func(http.ResponseWriter, *http.Request) {
+	var service serviceHandlerImpl
+	var resWriter = ResultWriter{}
+	if isJson {
+		service = serviceHandlerImpl{
+			builder:     responsebuilder.New(json.NewEncoder(&resWriter)),
+			contentType: contentJson,
+			resWriter:   &resWriter,
+		}
+	} else {
+		service = serviceHandlerImpl{
+			builder:     responsebuilder.New(xml.NewEncoder(&resWriter)),
+			contentType: contentXml,
+			resWriter:   &resWriter,
+		}
+	}
```

Comment:
Nitpick: this function does 2 things, render by json and render by xml, can be separate into 2 smaller functions for readability